### PR TITLE
Feat: Validation webhook for multicluster.x-k8s.io.v1alpha1.works

### PR DESCRIPTION
**Pull request by Ealianis**
### Description of your changes

Added admission chain validation webhook for multicluster.x-k8s.io.v1alpha1.works.

I have:
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested
Todo

### CODE REVIEW

File: deploy/admission-webhooks/validatingWebhookConfiguration.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          - multicluster.x-k8s.io
+        apiVersions:
+          - v1alpha1
+        operations:
+          - CREATE
+          - UPDATE
+        resources:
+          - works
+        scope: Namespaced
+    admissionReviewVersions:
```
Comment (Ealianis) [inline deploy/admission-webhooks/validatingWebhookConfiguration.yaml:36]: Fix versions

File: deploy/admission-webhooks/validatingWebhookConfiguration.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      - apiGroups:
+          - multicluster.x-k8s.io
+        apiVersions:
+          - v1alpha1
+        operations:
+          - CREATE
+          - UPDATE
+        resources:
+          - works
+        scope: Namespaced
```
Comment (Ealianis) [inline deploy/admission-webhooks/validatingWebhookConfiguration.yaml:35]: Investigate

File: webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	workv1alpha1 "sigs.k8s.io/work-api/pkg/apis/v1alpha1"
+)
+
+const (
+	validWorkRequest = "Valid Work request."
+)
+
+// +kubebuilder:webhook:path=/validate-v1-pod,mutating=false,failurePolicy=fail,groups="",resources=pods,verbs=create;update,versions=v1,name=vpod.kb.io
```
Comment (Ealianis) [inline webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go:17]: Fix kubebuilder annotation.

File: webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// Handle the validation of Work resources.
+func (v *WorkValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	work := &workv1alpha1.Work{}
+
+	err := v.decoder.Decode(req, work)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	// ToDo - Validation of Work object
```
Comment (Ealianis) [inline webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go:34]: Need AC on validation. Discuss with @ryanzhang-oss .

File: deploy/admission-webhooks/validatingWebhookConfiguration.yaml (hunk 2 above)
Comment (Ealianis) [reply deploy/admission-webhooks/validatingWebhookConfiguration.yaml:35]: Work may reside in multiple namespaces. Thus use *.
https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#configure-admission-webhooks-on-the-fly

File: webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package work
```
Comment (ryanzhang-oss) [inline webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go:1]: please move this under pkg

**Review by ryanzhang-oss — COMMENTED** [review]
what is implemented? I don't seem to see a single validation logic.

File: webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go (hunk 5 above)
Comment (Ealianis) [reply webhooks/multicluster.x-k8s.io/v1alpha1/work/validating_handler.go:1]: Will do.

**Comment by Ealianis** [comment]
Closing PR as it was decided that the Work validation webhook is not yet needed.

**Pull request closed without merging**